src/uu/pathchk/src/pathchk.rs

```diff
@@ -63,19 +62,14 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
 
     // free strings are path operands
     // FIXME: TCS, seems inefficient and overly verbose (?)
-    let mut res = true;
     for p in paths.unwrap() {
         let mut path = Vec::new();
         for path_segment in p.split('/') {
             path.push(path_segment.to_string());
         }
-        res &= check_path(&mode, &path);
+        check_path(&mode, &path)?;
```

Comment:
This now short circuits if an error occurs, is that the intended behaviour? Because if I try multiple bad paths in GNU, I get:
```
❯ pathchk "" ""
pathchk: '': No such file or directory
pathchk: '': No such file or directory
```
so it looks like it should not short-circuit. I think we can use the `show!` macro here, which will set the error code.

Also we should open an issue for the string handling here. It should operate on `Path` and `OsStr` instead of `String` I think.

Reply:
Good catch, thanks. You also discovered a bug in our implementation because `pathchk "" ""` doesn't fail.

Reply:
I made an issue for it!